File: aria-web/src/types/patient.ts
export interface Patient {
  id: string;
  medical_record_number: string;
  first_name: string;
  last_name: string;
  date_of_birth?: string;
  gender?: 'M' | 'F' | 'O';
  phone?: string;
  address?: string;
  created_at: string;
}

export interface PatientCreate {
  medical_record_number: string;
  first_name: string;
  last_name: string;
  date_of_birth?: string;
  gender?: 'M' | 'F' | 'O';
  phone?: string;
  address?: string;
}

export interface PatientUpdate {
  first_name?: string;
  last_name?: string;
  date_of_birth?: string;
  gender?: 'M' | 'F' | 'O';
  phone?: string;
  address?: string;
}

export interface PatientListResponse {
  items: Patient[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

export interface PatientStatsResponse {
  total_patients: number;
  gender_distribution: {
    male: number;
    female: number;
    other: number;
  };
  patients_with_analyses: number;
  percentage_with_analyses: number;
}

File: app/app/app/schemas/patient.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional
import uuid

class PatientBase(BaseModel):
    medical_record_number: str
    first_name: str
    last_name: str
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None

class PatientCreate(PatientBase):
    pass

class PatientUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None

class PatientResponse(PatientBase):
    id: uuid.UUID
    created_at: datetime

    class Config:
        from_attributes = True

class PatientListResponse(BaseModel):
    items: list[PatientResponse]
    total: int
    page: int
    per_page: int
    pages: int
